packages/datadog-plugin-gcloud-pubsub/src/index.js

```diff
@@ -0,0 +1,76 @@
+'use strict'
+
+function createWrapRequest (tracer, config) {
+  return function wrapRequest (request) {
+    return function wrappedRequest (cfg = { reqOpts: {} }, cb) {
+      let topic
+      if (cfg.reqOpts) {
+        topic = cfg.reqOpts[cfg.method === 'createTopic' ? 'name' : 'topic']
+      }
+      const tags = {
+        component: 'google-cloud-pubsub',
+        'resource.name': topic || 'default',
+        'service.name': config.service || `${tracer._service}-pubsub`,
+        'pubsub.method': cfg.method,
+        'pubsub.projectid': this.projectId,
+        'pubsub.topic': topic
+      }
+      cb = tracer.scope().bind(cb)
+      return tracer.trace('gpubsub.request', { tags }, (span, done) => {
+        if (cfg.reqOpts && cfg.method === 'publish') {
+          for (const msg of cfg.reqOpts.messages) {
+            if (!msg.attributes) {
+              msg.attributes = {}
+            }
+            tracer.inject(span, 'text_map', msg.attributes)
+          }
+        }
+
+        arguments[arguments.length - 1] = function (err) {
```

Comment:
This leads to believe the callback index is dynamic. I would say to either hardcode here, or make it fully dynamic everywhere in the function.